sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -514,12 +524,13 @@ private[sql] object Expand {
  * @param projections to apply
  * @param output of all projections.
  * @param child operator.
+ * @param groupByAttrs the attributes used in group by.
  */
 case class Expand(
     projections: Seq[Seq[Expression]],
     output: Seq[Attribute],
-    child: LogicalPlan) extends UnaryNode {
-
+    child: LogicalPlan,
+    groupByAttrs: Seq[Attribute]) extends UnaryNode {
```

Comment:
Yeah, I do think it would be good if it just took a `Seq[Seq[NamedExpression]]` (or at least I can't come up with anything better).  I'd still consider breaking this into two PRs.  Simple fix for now that just removes invalid constraints and a refactoring that add back in valid ones.